### ISSUE TITLE
[BUG] Unable to draw circle shape

### ISSUE THREAD

**Issue by rexy712** [Kruptein/PlanarAlly#1387]
**Describe the bug**
When trying to draw a circle shape, selecting the draw tool, placing the starting point, all goes as expected. When trying to finish the shape drawing, I am unable to. Unclicking the left mouse button, right clicking, middle clicking, nothing completes the shape and the only way to escape the drawing state is by hitting the escape key, which cancels the shape drawing.

**To Reproduce**
1. Enter build mode, select draw tool and the circle shape brush.
2. Begin drawing a circle.
3. Attempt to complete the circle.

**Expected behavior**
In the past, I can't recall if it was releasing left click after dragging that completed it or if it was a right click (like the polygon brush). I expect when I finish dragging and then release left click it would complete.

**Screenshots**

https://github.com/Kruptein/PlanarAlly/assets/2442544/f1278266-e598-4111-9a90-6691ce5cf4a2

**Desktop:**
 - OS: gentoo linux
 - Browser: firefox 122.0, ungoogled-chromium 119.0.6054.199
 - Version: dev branch commit [a1517bb](https://github.com/Kruptein/PlanarAlly/commit/a1517bbeee758e25227f933c4344fd6586ac02b1)

**Comment by Kruptein** [comment]
Hm curious I don't have this behaviour on my end on that same commit.

Is there anything appearing in the browser or server console?

**Comment by rexy712** [comment]
I checked the server logs and there are no errors on that end.

The browser console shows this error when releasing the left click button:
```
vue.esm-browser.prod.js:6 TypeError: Cannot read properties of undefined (reading 'x')
    at Vu (Game-3HCm_-Hf.js:2:7257)
    at Yu (Game-3HCm_-Hf.js:2:6654)
    at Wa.resizeToGrid (Game-3HCm_-Hf.js:4:75526)
    at UM.onUp (Game-3HCm_-Hf.js:5:77016)
    at UM.onMouseUp (Game-3HCm_-Hf.js:5:63064)
    at Proxy.QT (Game-3HCm_-Hf.js:5:137066)
    at C.onMouseup.e.<computed>.e.<computed> (Game-3HCm_-Hf.js:83:230955)
    at on (vue.esm-browser.prod.js:6:17402)
    at rn (vue.esm-browser.prod.js:6:17480)
    at HTMLDivElement.n (vue.esm-browser.prod.js:8:46226)
```

**Comment by rexy712** [comment]
Also of note is that I actually can draw a circle under some circumstances.

If I hold Alt while drawing, it disables grid snapping and the circle brush works fine.

It also works if I don't hold Alt and draw a circle without moving the cursor. It creates a small circle that can then be correctly resized. I believe because it doesn't attempt to do any grid snapping.

**Comment by Kruptein** [comment]
Ah that's why I couldnt repro it, I have alt behaviour reversed by default, so yeah it's snapping related, thanks for the info

**Comment by Kruptein** [comment]
Closed in #1390
